[Pang Kuang Wei] iP

PangKuangWei opened the pull request:
# Duke

### A Bullet List that tells you that Duke is

* text-based
* easy to learn
* ~FAST~ **SUPER FAST** to use

### All you need to do is follow this Numbered List

1. Download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉

### If you Java programmer, you *can* use it to practice Java too. Here's the code block of the `main` method.

```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
### Features

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

### That Inspiring Quote

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Commit:
Add Gradle support

PangKuangWei committed:
Implemented Level-1

PangKuangWei committed:
Implemented Level-2

PangKuangWei committed:
Implmented Level-3

PangKuangWei committed:
Implemented Level-4

PangKuangWei committed:
Added input for A-TextUiTesting

PangKuangWei committed:
Implemented Level-5

PangKuangWei committed:
Implemented Level-6

PangKuangWei committed:
Implemented A-Enums

PangKuangWei committed:
Implement Level-7

PangKuangWei committed:
Implement Level-8

PangKuangWei committed:
Merge branch 'branch-Level-7'

PangKuangWei committed:
Merge branch 'branch-Level-8'

PangKuangWei committed:
Add Ui, Storage, Parser, TaskList classes

PangKuangWei committed:
Move all classes into duke package

PangKuangWei committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

PangKuangWei committed:
Add JUnit Tests

PangKuangWei committed:
Package the App as a JAR file

PangKuangWei committed:
Add JavaDoc comments

PangKuangWei committed:
Tweak the code to comply with a coding standard

PangKuangWei committed:
Implement Find

PangKuangWei committed:
Merge branch 'branch-A-JavaDoc'

PangKuangWei committed:
Merge branch 'branch-A-CodingStandard'

PangKuangWei committed:
Merge branch 'branch-Level-9'

PangKuangWei committed:
Add CheckStyle

HowSuen commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,91 @@
+     */
+    public static void parseCommand(String str, TaskList tasks) throws DukeException {
+        String[] splitStr = str.split(" ", 2);
+        switch (splitStr[0]) {
```
I like your use of `switch` statement here, it makes your code very neat and readable 😃

HowSuen commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,63 @@
+    private TaskList tasks;
+    private Ui ui;
+
+    enum TaskType {
```
I like your use of enums here! Would it also be possible to create a separate enum class so you need not call Duke.TaskType from other classes?

HowSuen commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,119 @@
+                    char taskType = data.charAt(0);
+                    switch (taskType) {
+                    case 'T':
+                        String[] todoSplit = data.split(" // ", 3);
```
Perhaps you can consider putting the logic for handling these cases in separate methods and simply call the methods for these different cases? Since the code for these 3 cases are somewhat similar, it may make your code less verbose and even simpler 😄

HowSuen commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,91 @@
+     * @return a LocalDateTime representing the date and time
+     * @throws DukeException if the given input is not in the specified format
+     */
+    public static LocalDateTime parseDateTime(String str) throws DukeException {
```
I think you can consider using `DateTimeFormatter` for handling these formats! 😉

HowSuen reviewed (commented):
Overall, I found your code easy to read and well-organised. The names and descriptions you gave were useful and easy to understand. You can consider dividing these classes into packages to better organise them. You may also consider splitting into more functions to make it less verbose and simpler.

PangKuangWei committed:
Implement Level-10 GUI using JavaFX

PangKuangWei committed:
Merge branch 'branch-A-CheckStyle'

PangKuangWei committed:
Merge branch 'branch-Level-10'

WuHaohui1231 commented on src/main/java/Duke.java:
```diff
@@ -23,9 +29,90 @@ public static void main(String[] args) {
             }
```
Perhaps there should be a `catch` block here

WuHaohui1231 commented on src/main/java/Task.java:
```diff
@@ -7,6 +7,8 @@ public Task(String description) {
     }
 
+    abstract public String getDataFormat();
+
```
Nice use of abstract method

WuHaohui1231 reviewed (commented):
Overall, the code is excellent. It follows the OOP principles

PangKuangWei committed:
Change bye command

Bye command only gives a reply and does not close the program.

Let's change the bye command so that it closes the program

PangKuangWei committed:
Add assertions

There are assumptions about the program state at different points in
the program. If they are wrong, it indicates a possible bug in the
code.

Let's use assert feature to define important assumptions that should
hold true at various points in the code.

PangKuangWei committed:
Improve code quality

Code has readability issues and unsafe practices.

Let's refer to the Code Quality topics and improve the code quality
where necessary.

PangKuangWei committed:
Merge pull request #1 from PangKuangWei/branch-A-Assertions

Add assertions

PangKuangWei committed:
Merge branch 'master' into branch-A-CodeQuality

PangKuangWei committed:
Merge pull request #2 from PangKuangWei/branch-A-CodeQuality

Improve code quality

PangKuangWei committed:
Set up Continuous Integration

PangKuangWei committed:
Add more natural date formats

Day inputs in the user command are not accepted.

Let's allow day inputs and interpret them as the date of the next
time the day is reached.

PangKuangWei committed:
Merge branch 'branch-C-NaturalDates'

PangKuangWei committed:
Add User Guide

PangKuangWei committed:
Fix documentation
